### PR TITLE
implement system messages UI

### PR DESCRIPTION
**Pull request by petrjasek**
SDESK-5969 SDESK-5659

### CODE REVIEW

File: scripts/core/superdesk-api.d.ts [hunk 1]
```diff
@@ -897,6 +898,9 @@ declare module 'superdesk-api' {
 
         field: string;
 
+        /** default value */
+        value?: string;
```
Comment (tomaskikutis) [inline scripts/core/superdesk-api.d.ts:902]: Is this used anywhere?

File: scripts/core/ui/components/generic-form/form-field.tsx [hunk 2]
```diff
@@ -43,9 +46,13 @@ export function getFormFieldPreviewComponent(
     item: {readonly [key: string]: any},
     formFieldConfig: IFormField,
     options: { showAsPlainText?: boolean } = {},
-): JSX.Element {
+): JSX.Element | null {
     const Component = getFormFieldComponent(formFieldConfig.type);
 
+    if (Component == null) {
```
Comment (tomaskikutis) [inline scripts/core/ui/components/generic-form/form-field.tsx:52]: Why would it be `null`?

File: scripts/core/ui/components/generic-form/from-group.tsx [hunk 3]
```diff
@@ -37,6 +37,10 @@ export class FormViewEdit extends React.Component<IProps> {
                         } else if (isIFormField(item)) {
                             const FieldComponent = getFormFieldComponent(item.type);
 
+                            if (FieldComponent == null) {
```
Comment (tomaskikutis) [inline scripts/core/ui/components/generic-form/from-group.tsx:40]: Why would it be `null`?

File: scripts/core/ui/components/generic-form/form-field.tsx (hunk 2 above)
Comment (petrjasek) [reply scripts/core/ui/components/generic-form/form-field.tsx:52]: I was adding a hidden field, which wouldn't be rendered, but at the end I'm not using it

Comment (tomaskikutis) [reply scripts/core/ui/components/generic-form/form-field.tsx:52]: I'm not getting it. How would a "hidden field" work here? From what I understand, all field types should have a corresponding component that would render the input field.

Comment (petrjasek) [reply scripts/core/ui/components/generic-form/form-field.tsx:52]: it was like `<input type="hidden" value="...">`, was using it for `user_id`, but then I've moved that handling to the server

File: scripts/apps/system-messages/index.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import {SystemMessagesSettingsComponent} from './components/system-messages-settings';
+import {SystemMessagesComponent} from './components/system-messages';
+import {IBaseRestApiResponse, IUser} from 'superdesk-api';
+
+export interface ISystemMessage extends IBaseRestApiResponse {
+    type: 'warning' | 'alert' | 'primary' | 'success';
+    is_active: boolean;
+    message_title: string;
+    message: string;
+    user_id: IUser['_id'];
```
[the remark this answers is not in the document]
Comment (petrjasek) [reply scripts/apps/system-messages/index.ts:12]: it's always populated by server

File: scripts/core/ui/components/generic-form/form-field.tsx (hunk 2 above)
Comment (tomaskikutis) [reply scripts/core/ui/components/generic-form/form-field.tsx:52]: I think `component_parameters` can be used for holding extra info that you tried to put inside a hidden field. Or if it doesn't work in some scenario, maybe it'd be worth adding "hidden field" input type. I'd drop this for now so we don't carry dead code that we would be afraid to remove in the future.

File: scripts/core/superdesk-api.d.ts (hunk 1 above)
Comment (petrjasek) [reply scripts/core/superdesk-api.d.ts:902]: I had it for that hidden field, can remove it now

**Review by tomaskikutis — APPROVED** [review]

**Pull request merged**